csrc/multidevice/pipeline.h

```diff
@@ -63,8 +63,11 @@ class PipelineStageDescriptor final {
   using ValSet = VectorOfUniqueEntries<Val*>;
 
  public:
-  PipelineStageDescriptor() : unique_id(running_unique_id_++) {}
+  PipelineStageDescriptor()
+      : auto_schedule(true), unique_id(running_unique_id_++) {}
 
+  // whether the stage should be auto-scheduled
+  bool auto_schedule;
```

Comment:
We could, but what is the reason to do so? `PipelineStageDescriptor` is supposed to be fully configurable and be passed as argument for creating PipelineStage. We could still set it in the constructor but it will just make the syntax of writing tests a bit less flexible.
I think in the future this class will be renamed `PipelineStageParams` and be made a `struct` rather than a `class` (as you once suggested). I think it is more convenient to let the members public and non-const so they can be manually set in the tests

Reply:
Ignoring the convenience of writing tests, can this be `const`? Making a const variable `const` is also helpful to understand the intention of the API design, which is why I asked this question.